react/features/gifs/components/web/GifsMenu.tsx

```diff
@@ -77,19 +83,25 @@ const useStyles = makeStyles()(theme => {
     };
 });
 
+interface IProps {
+    columns?: number;
+    parent: IReactionsMenuParent;
```

Comment:
This feels weird. Normally in React the child component shouldn't know about the parent component. Can't we just pass an `inDrawer` prop or something?

Reply:
As you can see from the type `parent` is not a component at all. `parent` is an enum that indicates where the component will be rendered - in drawer, in the overflow menu (not a drawer) or as part of a button.

In my initial implementation we had 2 props inDrawer and overflowMenu in order to establish where the component is rendered. If you check the discussion with @saghul from the previous comments I realized that this is not great and changed that to this enum.

The enum implementation is simpler and more readable.